pallets/staking/src/lib.rs

```diff
@@ -2665,10 +2721,31 @@ decl_module! {
             SlashingAllowedFor::put(slashing_switch);
             Self::deposit_event(RawEvent::SlashingAllowedForChanged(slashing_switch));
         }
+
+        /// Scale the intended validator count for a given DID.
+        ///
+        /// # Arguments
+        /// * origin which must be the required origin for adding a potential validator.
+        /// * identity to add as a validator.
+        /// * additional amount by which `intended_count` gets increased.
+        #[weight = 150_000_000]
+        pub fn scale_permissioned_validator_intended_count(origin, identity: IdentityId, additional: u32) -> DispatchResult {
```

Comment:
I think we should change `scale` to `update` as `scale` makes it sound like it is being adjusted by a multiplier.

Should we allow the `intended_count` to be decreased as well as increased? In the decrease case, it wouldn't impact running validators until they stop validating "naturally", but would stop additional validators being added later.

Reply:
This function allows to increase the intended count for a given permissioned identity, So I think they can simply not run the validators and save the cost of running this extrinsic.

Reply:
I can use `increase` instead of `scale`